src/spork.cpp

```diff
@@ -153,7 +153,7 @@ void CSporkManager::ProcessSpork(CNode* pfrom, std::string& strCommand, CDataStr
         std::map<SporkId, CSporkMessage>::iterator it = mapSporksActive.begin();
 
         while (it != mapSporksActive.end()) {
-            pfrom->PushMessage(NetMsgType::SPORK, it->second);
+            g_connman->PushMessage(pfrom, NetMsgType::SPORK, it->second);
```

Comment:
This one is a tricky one, the node could be sending the sporks before and after receiving the version message.

Need to think further on how to fix it, the ´GetSendVersion´ function is not atomic but almost.. make it atomic could be an overkilling and only call the method without guard could be too naive.

Reply:
not valid anymore, we can get rid off the `INIT_PROTO_VERSION ` in every spork message verifying to only send messages to initialized peers in the masternode-sync loop and moving/removing the `getsporks` message from the process version message block. 
https://github.com/PIVX-Project/PIVX/pull/1835#discussion_r483322072 .

resolving comments.